commit 92b013bde301bf9d4672e3cd866e0f0c75f983df
Author:     Yuhong Wen <yuhongw@nvidia.com>
AuthorDate: Thu Sep 1 00:49:42 2022 -0400
Commit:     GitHub <noreply@github.com>
CommitDate: Thu Sep 1 00:49:42 2022 -0400

    Simulator (#757)
    
    * Added SimulatorServer and client register.
    
    * Extracted ServerAppRunner and ClientAppRunner.
    
    * refactoried.
    
    * multiple clients.
    
    * created multiple clients, run the client in turn.
    
    * Execute the training for clients in turn, and init the clientRunner as needed.
    
    * clean up previous client before run.
    
    * Added Simulator logger.
    
    * Skipped the current running client when getting the next client to run.
    
    * Refactored.
    
    * keep to the same client when number of client the same as the threads number.
    
    * optimized.
    
    * Refactored.
    
    * refactored.
    
    * cleaned up.
    
    * refactored.
    
    * Added unittest.
    
    * Moved the run_client_thread into SimulatorRunner.
    
    * Added the client_list names option.
    
    * Added client_name file option.
    
    * Made simulator easy to debug.
    
    * Enabled the single app simulator run.
    
    * if no client names provided, extract the client list from the meta.json.
    
    * Added unittest for simulator_runner.
    
    * Added more unittests.
    
    * renamed unittests.
    
    * refactoried.
    
    * Fixed isort.
    
    * Removed no need import.
    
    * fixed isort.
    
    * refactored.
    
    * WIP: preepare for multi-gpu client child process.
    
    * Separate the SimulatorClientRunner, prepare for client child process.
    
    * Added support to multi-GPU run support.
    
    * log errors when client config error.
    
    * Added custom folder for client_runner_process.
    
    * Enhanced the simulator to enable relative path.
    
    * Added the log.txt for simulator run.
    
    * Added gpu options.
    
    * Added option error check for self.args.gpu or self.args.threads.
    
    * Refactored.
    
    * adjust max_num_clients.
    
    * create a separate process when running the client task.
    
    * cleaned up.
    
    * enabled to sub-process to run the client training, solving the GPU memory release issue.
    
    * refactored.
    
    * Changed the -t to -p.
    
    * Added indentity to the log.txt.
    
    * Added SimulatorConstants.
    
    * Ensure proper stop.
    
    * Changed the sys.exit(-1) to return False.
    
    * Refactored to use SimulatorClientAppRunner.
    
    * Run the multi-gpu with multi-threads.
    
    * update comment.
    
    * No neeed for ports in the SimulatorDeployer __init__() any more.
    
    * Changed the default client names format.
    
    * code style.
    
    * Fixed issues raised by QA.
    
    * Addressed QA issues.
    
    * Fixed unittest.
---
 nvflare/private/fed/server/server_engine.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/nvflare/private/fed/server/server_engine.py b/nvflare/private/fed/server/server_engine.py
index b4cd05d8..346ee312 100644
--- a/nvflare/private/fed/server/server_engine.py
+++ b/nvflare/private/fed/server/server_engine.py
@@ -14,7 +14,6 @@
 
 import copy
 import logging
-import multiprocessing
 import os
 import pickle
 import re
@@ -120,7 +119,7 @@ class ServerEngine(ServerEngineInternalSpec):
         self.parent_conn_lock = Lock()
         self.job_runner = None
         self.job_def_manager = None
-        self.snapshot_lock = multiprocessing.Lock()
+        # self.snapshot_lock = multiprocessing.Lock()
 
     def _get_server_app_folder(self):
         return WorkspaceConstants.APP_PREFIX + "server"
